### PR TITLE
added filter for catalog

### PR DESCRIPTION
**Pull request by andrew13pol**

### CODE REVIEW

File: src/main/java/com/zufar/icedlatte/product/endpoint/ProductsEndpoint.java [hunk 1]
```diff
@@ -67,4 +70,10 @@ public ResponseEntity<List<ProductInfoDto>> getProductsByIds(@RequestBody final
         return ResponseEntity.ok()
                 .body(products);
     }
+
```
Comment (M437A) [inline src/main/java/com/zufar/icedlatte/product/endpoint/ProductsEndpoint.java:73]: add open api description and make this method "Override"

File: src/main/java/com/zufar/icedlatte/product/endpoint/ProductsEndpoint.java [hunk 2]
```diff
@@ -67,4 +70,10 @@ public ResponseEntity<List<ProductInfoDto>> getProductsByIds(@RequestBody final
...
+    @GetMapping("/filter")
+    public ResponseEntity<List<ProductInfoDto>> getProductFilter(@RequestBody final ProductFilterDto productFilterDto){
+        List<ProductInfoDto> products = productFilterList.getProductFilterList(productFilterDto);
```
Comment (M437A) [inline src/main/java/com/zufar/icedlatte/product/endpoint/ProductsEndpoint.java:76]: Add a couple of logs, as in the methods above

File: src/main/java/com/zufar/icedlatte/filter/api/ProductFilterList.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+@Service
+@RequiredArgsConstructor
+public class ProductFilterList {
+    private final ProductInfoRepository productInfoRepository;
+    private final ProductInfoDtoConverter productInfoDtoConverter;
+    private final List<ProductFilter>productFilters;
```
Comment (M437A) [inline src/main/java/com/zufar/icedlatte/filter/api/ProductFilterList.java:21]: add space between  "List<ProductFilter>" and "productFilters"

File: src/main/java/com/zufar/icedlatte/filter/api/ProductFilterList.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
```
Comment (M437A) [inline src/main/java/com/zufar/icedlatte/filter/api/ProductFilterList.java:22]: add an abstract of the transaction

File: src/main/java/com/zufar/icedlatte/filter/api/ProductFilterList.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    private List<ProductInfoDto> applyFilter(Stream<ProductInfo> productInfoStream, ProductFilterDto productFilterDto) {
+        List<ProductInfo> productInfo = new ArrayList<>();
+        List<ProductFilter> requiredFilters = productFilters.stream()
+                .filter(filter -> filter.isApplicable(productFilterDto))
+                .toList();
+        for (ProductFilter filter : requiredFilters) {
+            productInfo = filter.apply(productInfoStream, productFilterDto);
+        }
+        return productInfo.stream().map(productInfoDtoConverter::toDto).toList();
```
Comment (M437A) [inline src/main/java/com/zufar/icedlatte/filter/api/ProductFilterList.java:37]: not necessarily, but I prefer to do it line by line

File: src/main/java/com/zufar/icedlatte/filter/product_filter/ProductFilterMaxPrice.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package com.zufar.icedlatte.filter.product_filter;
+
+import com.zufar.icedlatte.filter.ProductFilter;
+import com.zufar.icedlatte.filter.dto.ProductFilterDto;
+import com.zufar.icedlatte.product.entity.ProductInfo;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public class ProductFilterMaxPrice implements ProductFilter {
```
Comment (M437A) [inline src/main/java/com/zufar/icedlatte/filter/product_filter/ProductFilterMaxPrice.java:10]: it's must be component

File: src/main/java/com/zufar/icedlatte/filter/product_filter/ProductFilterMinPrice.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+public class ProductFilterMinPrice implements ProductFilter {
+
+    @Override
+    public boolean isApplicable(ProductFilterDto productFilterDto) {
+        return productFilterDto.getMinPrice() != null;
+    }
+
+        @Override
+        public List<ProductInfo> apply(Stream<ProductInfo> productStream, ProductFilterDto productFilterDto) {
```
Comment (annstriganova) [inline src/main/java/com/zufar/icedlatte/filter/product_filter/ProductFilterMinPrice.java:18]: nit: inconsistent indentation, use `option + cmd + L` to format

File: src/main/java/com/zufar/icedlatte/filter/product_filter/ProductFilterMinPrice.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
+package com.zufar.icedlatte.filter.product_filter;
```
Comment (annstriganova) [inline src/main/java/com/zufar/icedlatte/filter/product_filter/ProductFilterMinPrice.java:1]: I think that all this logic should be located in `com.zufar.icedlatte.product.filter` package. 

Also, please note that the snake_case is not common in Java

File: src/main/java/com/zufar/icedlatte/product/endpoint/ProductsEndpoint.java (hunk 1 above)
Comment (annstriganova) [reply src/main/java/com/zufar/icedlatte/product/endpoint/ProductsEndpoint.java:73]: Just in case: `src/main/resources/api-specs/product-openapi.yaml`

**Review by annstriganova — CHANGES_REQUESTED** [review]
Could you please also add a couple of tests for the new endpoint? 👀

**Pull request closed without merging**